aten/src/ATen/xpu/XPUEvent.h

```diff
@@ -0,0 +1,136 @@
+#pragma once
+#include <ATen/xpu/XPUContext.h>
+#include <c10/util/Optional.h>
+
+namespace at::xpu {
+
+/*
+ * XPUEvent are movable not copyable wrappers around SYCL event. XPUEvent are
+ * constructed lazily when first recorded. It has a device, and this device is
+ * acquired from the first recording stream. Later streams that record the event
+ * must match the same device.
+ */
+struct TORCH_XPU_API XPUEvent {
+  // Constructors
+  XPUEvent() noexcept = default;
+  XPUEvent(bool enable_timing) noexcept : enable_timing_{enable_timing} {}
+
```

Comment:
The XPUEvent is incomplete, because of lack of device guard?

Reply:
device guard is not necessary. XPUEvent currently does not support `IPC`.

Reply:
SYCL already has IPC API now?

Reply:
No. SYCL doesn't have, L0 has.

Reply:
What's our plan of supporting the IPC event? If we do not support it right now, suggest to add a note in the code explaining the reasons and impact.

Reply:
Yes. We should add note for it. @guangyey